### PR TITLE
Add cache metrics docs

### PR DESCRIPTION
**Pull request by rmccorm4**
Documentation on new cache metrics/stats introduced in https://github.com/triton-inference-server/server/pull/3569

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -54,14 +54,37 @@ The following table describes the available metrics.
...
+
+## Response Cache
+
+Compute latency metrics in the table above are calculated for the
+time spent in model inference backends. If the response cache is enabled for a
+given model (and a non-zero `--response-cache-byte-size` is defined for the
+server), total inference times may be affected by response cache lookup times.
+
+On cache hits, the cached response will be used instead of going through the
+inference backend, so compute times will not be recorded.
```
Comment (deadeyegoodwin) [inline docs/metrics.md:84]: Perhaps say that "Cache Hit Lookup Time" indicates time spent looking up the response and "Compute Input Time", "Compute Time" and "Compute Output Time" are not recorded.

And then the reverse for cache misses.

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: docs/metrics.md [hunk 2]
```diff
@@ -54,14 +54,37 @@ The following table describes the available metrics.
...
+|              |Total Cache Utilization |Response Cache utilization rate (0.0 - 1.0) |Server-wide |Per second |
+|              |Cache Hit Count |Number of response cache hits per model |Per model |Per request |
+|              |Cache Hit Lookup Time |Cumulative time requests spend retrieving a cached response per-model on cache hits, does not include cache misses (microseconds) |Per model |Per request |
+
+## Response Cache
+
+Compute latency metrics in the table above are calculated for the
+time spent in model inference backends. If the response cache is enabled for a
+given model (and a non-zero `--response-cache-byte-size` is defined for the
+server), total inference times may be affected by response cache lookup times.
```
Comment (tanmayv25) [inline docs/metrics.md:81]: You can link response cache doc here and prevent repetition here.

File: docs/metrics.md [hunk 3]
```diff
@@ -54,14 +54,37 @@ The following table describes the available metrics.
...
+|              |Compute Output Time|Cumulative time requests spend processing inference outputs (in the framework backend, does not include cached requests)     |Per model  |Per request  |
+|Response Cache|Total Cache Entry Count |Total number of responses stored in response cache across all models |Server-wide |Per second |
+|              |Total Cache Lookup Count |Total number of response cache lookups done by Triton across all models |Server-wide |Per second |
+|              |Total Cache Hit Count |Total number of response cache hits across all models |Server-wide |Per second |
+|              |Total Cache Miss Count |Total number of response cache misses across all models |Server-wide |Per second |
+|              |Total Cache Eviction Count |Total number of response cache evictions across all models |Server-wide |Per second |
+|              |Total Cache Lookup Time |Cumulative time requests spend checking for a cached response across all models (microseconds) |Server-wide |Per second |
+|              |Total Cache Utilization |Response Cache utilization rate (0.0 - 1.0) |Server-wide |Per second |
```
Comment (tanmayv25) [inline docs/metrics.md:72]: Doesn't utilization of 0.0-100.0 make more sense?

Comment (rmccorm4) [reply docs/metrics.md:72]: I wanted to keep it consistent with the existing `GPU Utilization rate (0.0 - 1.0)` metric

**Review by deadeyegoodwin — APPROVED** [review]

**Comment by rmccorm4** [comment]
Holding off for perf analyzer review and testing

**Pull request merged**